utils/utils.go

```diff
@@ -181,3 +183,44 @@ func ExpandVariables(vars map[string]string, str string) string {
 	// if no variables are expanded return the original string
 	return str
 }
+
+// IsVirtualBox returns true if the running system is executed
+// from within VirtualBox
+// Attempt to parse the System Management BIOS (SMBIOS) and
+// Desktop Management Interface (DMI) to determine if we are
+// executing inside a VirtualBox. Ignoring error conditions and
+// assuming we are not VirtualBox.
+func IsVirtualBox() bool {
+	virtualBox := false
+
+	// Find SMBIOS data in operating system-specific location.
+	rc, _, err := smbios.Stream()
```

Comment:
are you sure there isn't a case where rc will not require a call for rc.Close() even if we got an err? I would suggest moving the deferred rc.Close() up here.